packages/mui-base/src/NumberInputUnstyled/NumberInputUnstyled.types.ts

```diff
@@ -0,0 +1,52 @@
+import { OverrideProps, Simplify } from '@mui/types';
+import { UseNumberInputParameters } from './useNumberInput.types';
+import { SlotComponentProps } from '../utils';
+
+export interface NumberInputUnstyledComponentsPropsOverrides {}
+
+export type NumberInputUnstyledOwnProps = {} & Omit<UseNumberInputParameters, 'error'> & {
+    /**
+     * The id of the `input` element.
+     */
+    id?: string;
+    /**
+     * The props used for each slot inside the Input.
+     * @default {}
+     */
+    slotProps?: {
+      input?: SlotComponentProps<
+        'input',
+        NumberInputUnstyledComponentsPropsOverrides,
+        NumberInputUnstyledOwnerState
+      >;
+    };
```

Comment:
```suggestion
    slotProps?: {
      root?: SlotComponentProps<
        'root',
        NumberInputUnstyledComponentsPropsOverrides,
        NumberInputUnstyledOwnerState
      >;
    };
```

@michaldudak should we use `root` as slot if there is only one slot in the component?

Reply:
Yes, we use `root` whenever there's just one slot. But the first generc parameter of SlotComponentProps should remain as `input` - it's the default component.

```suggestion
    slotProps?: {
      root?: SlotComponentProps<
        'input',
        NumberInputUnstyledComponentsPropsOverrides,
        NumberInputUnstyledOwnerState
      >;
    };
```

Reply:
Got it! For this I will have one root containing 3 other slots